src/server/stream_family.cc

```diff
@@ -1272,6 +1272,38 @@ OpResult<uint32_t> OpDel(const OpArgs& op_args, string_view key, absl::Span<stre
   return deleted;
 }
 
+// XACK key groupname id [id ...]
+OpResult<uint32_t> OpAck(const OpArgs& op_args, string_view key, string_view gname,
+                         absl::Span<streamID> ids) {
+  OpResult<pair<stream*, streamCG*>> res = FindGroup(op_args, key, gname);
+  if (!res)
+    return res.status();
+  auto [stream_inst, cg] = *res;
+
+  if (cg == nullptr || stream_inst == nullptr) {
+    return 0;
```

Comment:
Originally, I wrote this long message here cause I thought you were transforming an error to a valid result. Upon testing this on redis (and going over the ACK docs) it seems to be correct. Therefore, both if the cg and the key does not exist we should return 0.